### PR TITLE
Workaround for having 2 IPs on an interface at start

### LINKED ISSUE #729: Vagrant CoreOS demo fails
Running through this doc: 
http://docs.projectcalico.org/v2.2/getting-started/kubernetes/installation/vagrant/

Initially, things seem to be working fine:
```
core@k8s-master ~ $ ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 08:00:27:13:3a:0c brd ff:ff:ff:ff:ff:ff
    inet 10.0.2.15/24 brd 10.0.2.255 scope global dynamic eth0
       valid_lft 85688sec preferred_lft 85688sec
    inet6 fe80::a00:27ff:fe13:3a0c/64 scope link 
       valid_lft forever preferred_lft forever
3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 08:00:27:ee:32:8c brd ff:ff:ff:ff:ff:ff
    inet 192.168.56.101/24 brd 192.168.56.255 scope global dynamic eth1
       valid_lft 479sec preferred_lft 479sec
    inet 172.18.18.101/24 brd 172.18.18.255 scope global eth1
       valid_lft forever preferred_lft forever
    inet6 fe80::a00:27ff:feee:328c/64 scope link 
       valid_lft forever preferred_lft forever
4: docker0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default 
    link/ether 02:42:43:e9:55:cd brd ff:ff:ff:ff:ff:ff
    inet 172.17.0.1/16 scope global docker0
       valid_lft forever preferred_lft forever
5: tunl0@NONE: <NOARP,UP,LOWER_UP> mtu 1440 qdisc noqueue state UNKNOWN group default qlen 1
    link/ipip 0.0.0.0 brd 0.0.0.0
    inet 192.168.235.192/32 scope global tunl0
       valid_lft forever preferred_lft forever
core@k8s-master ~ $ sudo ./calicoctl node status
Calico process is running.
IPv4 BGP status
+----------------+-------------------+-------+----------+-------------+
|  PEER ADDRESS  |     PEER TYPE     | STATE |  SINCE   |    INFO     |
+----------------+-------------------+-------+----------+-------------+
| 192.168.56.102 | node-to-node mesh | up    | 11:46:40 | Established |
| 192.168.56.103 | node-to-node mesh | up    | 11:46:39 | Established |
+----------------+-------------------+-------+----------+-------------+
IPv6 BGP status
No IPv6 peers found.
```
But later, it looks like the node loses the IP that calico is using for BGP:
```
core@k8s-master ~ $ sudo ./calicoctl node status
Calico process is running.
IPv4 BGP status
+----------------+-------------------+-------+----------+--------------------------------+
|  PEER ADDRESS  |     PEER TYPE     | STATE |  SINCE   |              INFO              |
+----------------+-------------------+-------+----------+--------------------------------+
| 192.168.56.102 | node-to-node mesh | start | 12:00:18 | Active Socket: Connection      |
|                |                   |       |          | closed                         |
| 192.168.56.103 | node-to-node mesh | start | 12:00:55 | Active Socket: Connection      |
|                |                   |       |          | closed                         |
+----------------+-------------------+-------+----------+--------------------------------+
IPv6 BGP status
No IPv6 peers found.
core@k8s-master ~ $ ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 08:00:27:13:3a:0c brd ff:ff:ff:ff:ff:ff
    inet 10.0.2.15/24 brd 10.0.2.255 scope global dynamic eth0
       valid_lft 84865sec preferred_lft 84865sec
    inet6 fe80::a00:27ff:fe13:3a0c/64 scope link 
       valid_lft forever preferred_lft forever
3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 08:00:27:ee:32:8c brd ff:ff:ff:ff:ff:ff
    inet 172.18.18.101/24 brd 172.18.18.255 scope global eth1
       valid_lft forever preferred_lft forever
    inet6 fe80::a00:27ff:feee:328c/64 scope link 
       valid_lft forever preferred_lft forever
4: docker0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default 
    link/ether 02:42:43:e9:55:cd brd ff:ff:ff:ff:ff:ff
    inet 172.17.0.1/16 scope global docker0
       valid_lft forever preferred_lft forever
5: tunl0@NONE: <NOARP,UP,LOWER_UP> mtu 1440 qdisc noqueue state UNKNOWN group default qlen 1
    link/ipip 0.0.0.0 brd 0.0.0.0
    inet 192.168.235.192/32 scope global tunl0
       valid_lft forever preferred_lft forever
```

### PR DESCRIPTION
**Pull request by tmjd**
Problem is with VirtualBox and its default DCHP server
Fixes #729

### CODE REVIEW

**Comment by tmjd** [comment]
I was able to reproduce this problem by:
1. destroying all my vagrant machines
2. removing all the `vboxnet[0-9]` interfaces that `ip a` reported
3. doing `vagrant up` with the unchanged Vagrantfile and cloud configs
4. Check `ip a` for each VM once `vagrant up` returned.
    - See multiple ip addresses reported for the eth1 interface

With the changes in the PR, doing the same steps as above the eth1 interface only had the expected IP address as set in the Vagrantfile.

**Comment by robbrockbank** [comment]
I still can't get this to work.  I wasn't hitting the problem with the docker coreOS demo, but I was with the rkt demo.  I added the same incantation to the rkt demo to no avail.

ifconfig down and up on didn't seem to do anything when I was ssh'd into calico-01

```
core@calico-01 ~ $ sudo ifconfig  eth1 down
core@calico-01 ~ $ sudo ifconfig  eth1 up  
core@calico-01 ~ $ ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 08:00:27:2f:c5:52 brd ff:ff:ff:ff:ff:ff
    inet 10.0.2.15/24 brd 10.0.2.255 scope global dynamic eth0
       valid_lft 86015sec preferred_lft 86015sec
    inet6 fe80::a00:27ff:fe2f:c552/64 scope link 
       valid_lft forever preferred_lft forever
3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 08:00:27:62:bd:e9 brd ff:ff:ff:ff:ff:ff
    inet 192.168.56.105/24 brd 192.168.56.255 scope global dynamic eth1
       valid_lft 809sec preferred_lft 809sec
    inet 172.18.18.101/24 brd 172.18.18.255 scope global eth1
       valid_lft forever preferred_lft forever
    inet6 fe80::a00:27ff:fe62:bde9/64 scope link 
       valid_lft forever preferred_lft forever
```

output from host:

```
vboxnet0: flags=8943<UP,BROADCAST,RUNNING,PROMISC,SIMPLEX,MULTICAST> mtu 1500
	ether 0a:00:27:00:00:00 
	inet 172.18.18.1 netmask 0xffffff00 broadcast 172.18.18.255
vboxnet1: flags=8842<BROADCAST,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	ether 0a:00:27:00:00:01 
vboxnet2: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	ether 0a:00:27:00:00:02 
	inet 172.17.8.1 netmask 0xffffff00 broadcast 172.17.8.255
vboxnet3: flags=8842<BROADCAST,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	ether 0a:00:27:00:00:03 
vboxnet4: flags=8842<BROADCAST,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	ether 0a:00:27:00:00:04 
```

**Comment by robbrockbank** [comment]
With the docker coreOS demo, it uses a different 172 subnet:

```
core@calico-01 ~ $ ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 08:00:27:2f:c5:52 brd ff:ff:ff:ff:ff:ff
    inet 10.0.2.15/24 brd 10.0.2.255 scope global dynamic eth0
       valid_lft 86256sec preferred_lft 86256sec
    inet6 fe80::a00:27ff:fe2f:c552/64 scope link 
       valid_lft forever preferred_lft forever
3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 08:00:27:6b:12:db brd ff:ff:ff:ff:ff:ff
    inet 172.17.8.101/24 brd 172.17.8.255 scope global eth1
       valid_lft forever preferred_lft forever
    inet6 fe80::a00:27ff:fe6b:12db/64 scope link 
       valid_lft forever preferred_lft forever
4: docker0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default 
    link/ether 02:42:49:0f:87:0d brd ff:ff:ff:ff:ff:ff
    inet 172.18.0.1/16 scope global docker0
       valid_lft forever preferred_lft forever
```

**Comment by robbrockbank** [comment]
As suggested by @tmjd, I tried adding the following to the vagrant files:

```
      host.vm.provision :shell, :inline => "sudo /usr/bin/ip addr flush dev eth1"
      host.vm.provision :shell, :inline => "sudo /usr/bin/ip addr add #{ip}/24 dev eth1"
```

which seems to do the trick nicely.

File: master/getting-started/rkt/installation/vagrant-coreos/Vagrantfile [hunk 1]
```diff
@@ -28,6 +28,9 @@ Vagrant.configure("2") do |config|
     config.vm.define vm_name do |host|
       host.vm.hostname = vm_name
       host.vm.network :private_network, ip: "172.18.18.#{i+100}"
+      # Workaround VirtualBox issue where eth1 has 2 IP Addresses at startup
+      host.vm.provision :shell, :inline => "sudo /usr/bin/ip addr flush dev eth1"
+      host.vm.provision :shell, :inline => "sudo /usr/bin/ip addr add #{ip}/24 dev eth1"
```
Comment (robbrockbank) [inline master/getting-started/rkt/installation/vagrant-coreos/Vagrantfile]: ip is not defined here.

File: v2.2/getting-started/rkt/installation/vagrant-coreos/Vagrantfile [hunk 2]
```diff
@@ -28,6 +28,9 @@ Vagrant.configure("2") do |config|
     config.vm.define vm_name do |host|
       host.vm.hostname = vm_name
       host.vm.network :private_network, ip: "172.18.18.#{i+100}"
+      # Workaround VirtualBox issue where eth1 has 2 IP Addresses at startup
+      host.vm.provision :shell, :inline => "sudo /usr/bin/ip addr flush dev eth1"
+      host.vm.provision :shell, :inline => "sudo /usr/bin/ip addr add #{ip}/24 dev eth1"
```
Comment (robbrockbank) [inline v2.2/getting-started/rkt/installation/vagrant-coreos/Vagrantfile]: ip not defined.

**Comment by robbrockbank** [comment]
LGTM.

@tmjd Are you needing me to test this before merge (which may not happen until later)

**Comment by tmjd** [comment]
@robbrockbank I just brought up each of those Vagrant environments and checked that the eth1 devices had the one expected IP address.

**Pull request merged**